File: algorithms/1_easy/C/02068_check_whether_two_strings_are_almost_equivalent/solution.py
class Solution:
    def checkAlmostEquivalent(self, word1: str, word2: str) -> bool:
        freqs = [0] * 26

        for ch in word1:
            freqs[ord(ch) - ord('a')] += 1

        for ch in word2:
            freqs[ord(ch) - ord('a')] -= 1

        for freq in freqs:
            if abs(freq) > 3:
                return False

        return True

File: algorithms/1_easy/C/02068_check_whether_two_strings_are_almost_equivalent/solution.cpp
class Solution {
public:
    bool checkAlmostEquivalent(string word1, string word2) {
        array<int, 26> freqs{};

        for (const char &ch: word1) {
            freqs[static_cast<size_t>(ch - 'a')] += 1;
        }

        for (const char &ch: word2) {
            freqs[static_cast<size_t>(ch - 'a')] -= 1;
        }

        for (const int &freq: freqs) {
            if (abs(freq) > 3) {
                return false;
            }
        }

        return true;
    }
};
